platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/synchronization/LearningSynchronizer.java

```diff
@@ -149,12 +150,15 @@ public void synchronize() throws InterruptedException {
      * Attempt to free any and all resources that were acquired during the reconnect attempt.
      */
     private void abort() {
-        logger.warn(RECONNECT.getMarker(), "deleting partially constructed tree");
+        logger.warn(
+                RECONNECT.getMarker(),
+                "Deleting partially constructed tree:\n{}",
+                new MerkleTreeVisualizer(newRoot).setDepth(5).render());
         try {
             if (newRoot != null) {
                 newRoot.release();
             }
-        } catch (final Exception ex) {
+        } catch (final Throwable ex) {
```

Comment:
This is a bit dangerous.  Might be better to catch RuntimeException and not Throwable, so Errors still exit the JVM (as they should).

Reply:
I will probably revert this change and will not include it to the PR. More important is to print the tree as a part of warn() call right above.